app/components/PickBakerStakeAmount.tsx

```diff
@@ -24,6 +36,38 @@ interface Props {
     hasPendingChange?: boolean;
 }
 
+/**
+ * Check whether the proposed stake will cause the account to breach the capital bound.
+ */
+function useCapitalBoundCheck(
+    accountInfo: AccountInfo | undefined,
+    stake: string
+) {
+    const rewardStatus = useAsyncMemo(getRewardStatusLatest);
+    const capitalBound = useCapitalBound();
+
+    if (
+        !capitalBound ||
+        !isValidCcdString(stake) ||
+        !rewardStatus ||
+        !isRewardStatusV1(rewardStatus)
+    ) {
+        return { showWarning: false, limitAfterUpdate: 0n };
```

Comment:
You can type this so that `limitAfterUpdate` only has to have a value if `showWarning === true`. Then you only have to check for `showWarning` when using it. Also it improves readability as `limitAfterUpdate` is not really `0` in this case.